numpy/distutils/ccompiler.py

```diff
@@ -620,7 +640,19 @@ def matcher(version_string):
             version = m.group('version')
             return version
 
-    status, output = exec_command(version_cmd, use_tee=0)
+    try:
+        output = subprocess.check_output(version_cmd)
+    except subprocess.CalledProcessError as exc:
+        output = exc.output
+        status = exc.returncode
+    except OSError:
+        # match the historical returns for a parent
+        # exception class caught by exec_command()
+        status = 127
+        output = b''
+    else:
+        output = filepath_from_subprocess_output(output)
```

Comment:
Strictly speaking this is not a filepath, but you're correct that this matches the old behavior. Add a comment to that effect, and then this looks good to me.